lib/components/Button/base/index.tsx

```diff
@@ -63,55 +64,70 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+                    {...rest}
+                >
+                    {iconPosition === "start" && icon && !isLoading && icon}
 
-                {isLoading && !disabled && (
-                    <LoaderIcon
-                        data-testid="button-loader"
-                        className="quill-button__loader-icon"
-                    />
-                )}
-                {children}
-                {label && !isLoading && (
-                    <span className="quill-button-label">
-                        {size === "sm" ? (
-                            <CaptionText color={color} bold>
-                                {label}
-                            </CaptionText>
-                        ) : (
-                            <Text size={labelSize} bold color={color}>
-                                {label}
-                            </Text>
-                        )}
-                    </span>
-                )}
-                {iconPosition === "end" && icon && !isLoading && icon}
-                {dropdown && DropdownIcon && !isLoading && (
-                    <DropdownIcon
-                        data-state={isDropdownOpen ? "open" : "close"}
-                        className={clsx(
-                            "quill-button__transform",
-                            isDropdownOpen && "quill-button__transform-rotate",
-                        )}
-                    />
-                )}
-            </button>
+                    {isLoading && !disabled && (
+                        <LoaderIcon
+                            data-testid="button-loader"
+                            className="quill-button__loader-icon"
+                        />
+                    )}
+                    {children}
+                    {label && !isLoading && (
+                        <span className="quill-button-label">
+                            {size === "sm" ? (
+                                <CaptionText color={color} bold>
+                                    {label}
+                                </CaptionText>
+                            ) : (
+                                <Text size={labelSize} bold color={color}>
+                                    {label}
+                                </Text>
+                            )}
+                        </span>
+                    )}
+                    {iconPosition === "end" && icon && !isLoading && icon}
+                    {dropdown && DropdownIcon && !isLoading && (
+                        <DropdownIcon
+                            data-state={isDropdownOpen ? "open" : "close"}
+                            className={clsx(
+                                "quill-button__transform",
+                                isDropdownOpen &&
+                                    "quill-button__transform-rotate",
+                            )}
+                        />
+                    )}
+                </button>
+            </Wrapper>
```

Comment:
i only wrapped the component with the Wrapper. Please check in github.dev using 'shift' + '.'